image.go

```diff
@@ -362,6 +471,85 @@ func (img *imageHDU) Write(data interface{}) error {
 	return err
 }
 
+// Image returns an image.Image value.
+func (imgHDU *imageHDU) Image() image.Image {
+
+	// The returned image.
+	var img image.Image
+
+	// Getting the HDU bitpix and axes.
+	header := imgHDU.Header()
+	bitpix := header.Bitpix()
+	axes := header.Axes()
+	raw := imgHDU.Raw()
+
+	// Image width and height.
+	w := axes[0]
+	h := axes[1]
+
+	switch bitpix {
+	case 8:
+		//Gray
+		min := image.Point{X: 0, Y: 0}
```

Comment:
(apologies, I swear I had posted these comments *before* going on holidays...)

`min` and `max` can be lifted out of the `switch`.
also, instead of creating the `image.Rectangle` the "pedestrian" way, you could create a `rect` variable like so:
```go
rect := image.Rect(0,0,w,h)
```
and drop the `min` and `max` variables (but keep the `rect` out of the `switch`)

finally, you could directly return the `img` created inside each `case`:

```go
case 8:
    img := image.NewGray(rect)
    img.Pix = raw
    return img
```

so the `var img image.Image` variable from line 478 could be dropped (and you can rename `imgHDU` as `img`)

one last thing.
the `image.NewGray`, `image.NewGray16`, `image.NewRGBA` and `image.NewRGBA64` functions create a value implementing `image.Image` but also allocate a `[]uint8` buffer with a size according to `min` and `max` values:
 https://golang.org/src/image/image.go?s=22661:22696#L813
This buffer is then discarded when we assign `raw` to it (`img.Pix = raw`)
This is wasteful and generates pressure on the garbage collector.

it would be best to create a "concrete" `image.Image` with the final, correct buffer directly:

```go
case 16:
    img := &image.Gray16{
        Pix: raw,
        Stride: 2*w,
        Rect: rect,
    }
    return img
```

and similarly for the others (including the `-32` and `-64` cases).

thanks.